components/server/src/api/workspace-service-api.ts

```diff
@@ -174,4 +185,107 @@ export class WorkspaceServiceAPI implements ServiceImpl<typeof WorkspaceServiceI
         response.workspace = this.apiConverter.toWorkspace(info);
         return response;
     }
+
+    async stopWorkspace(req: StopWorkspaceRequest): Promise<StopWorkspaceResponse> {
+        if (!req.workspaceId) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "workspaceId is required");
+        }
+        await this.workspaceService.stopWorkspace(ctxUserId(), req.workspaceId, "stopped via API");
+        const response = new StopWorkspaceResponse();
+        return response;
+    }
+
+    async deleteWorkspace(req: DeleteWorkspaceRequest): Promise<DeleteWorkspaceResponse> {
+        if (!req.workspaceId) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "workspaceId is required");
+        }
+        await this.workspaceService.deleteWorkspace(ctxUserId(), req.workspaceId, "user");
+        const response = new DeleteWorkspaceResponse();
+        return response;
+    }
+
+    async updateWorkspace(request: UpdateWorkspaceRequest): Promise<UpdateWorkspaceResponse> {
+        if (!request.workspaceId) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "workspaceId is required");
+        }
+        const userId = ctxUserId();
+
+        // check if user can access workspace first
+        await this.workspaceService.getWorkspace(userId, request.workspaceId);
+
+        const tasks: Array<Promise<any>> = [];
+
+        if (request.name) {
+            tasks.push(this.workspaceService.setDescription(userId, request.workspaceId, request.name));
+        }
+
+        if (request.admission) {
+            if (request.admission === AdmissionLevel.OWNER_ONLY) {
+                tasks.push(this.workspaceService.controlAdmission(userId, request.workspaceId, "owner"));
+            } else if (request.admission === AdmissionLevel.EVERYONE) {
+                tasks.push(this.workspaceService.controlAdmission(userId, request.workspaceId, "everyone"));
+            }
+        }
+
+        if (request.pinned !== undefined) {
+            tasks.push(this.workspaceService.setPinned(userId, request.workspaceId, request.pinned));
+        }
+        // TODO: Use all or allSettled? Because it does will update only parts of required changes
```

Comment:
Generally what should happen if one part throws should we revert other? Although update is partial should it be atomic?

I'm not sure how critical is to us, that sometimes it can be inconsistent. I'm not sure it is super hard to fix it though, it can be done with a follow up PR too. But we should create an issue at least if we think it is important property.

cc @csweichel @svenefftinge wdyt?

Reply:
@akosyakov  Could we just ignore this since we're doing migration now? Otherwise, what will we do if revert changes API failed?

Reply:
Or just add a new server API so that we can implement it atomically

Reply:
Yes, I said for the migration is not critical, but when we announce API it is important to understand what is our contract.

Reply:
no, it is fine do it later, just create an issue

Reply:
[issue](https://linear.app/gitpod/issue/EXP-983) created